src/parsers/rules.jl

```diff
@@ -11,6 +11,18 @@ struct Rule
     Rule(fn, priority, triggers="") = new(fn, priority, triggers)
 end
 
+# Two parsing rules are generally considered the same (for the purposes of enabling and
+# disabling them in the parser) if the types match --- the values on any fields do not
+# matter. In case this is not correct for a rule, the two-argument is_same_rule should be
+# appropriately overloaded. Some possible cases where this might be necessary:
+#   (1) A rule with type parameters, where even when the type parameter values are
```

Comment:
As a general note on this one, having rules that encode type parameters should be avoided where possible I think. I don't think any of the current set do, and from what I recall performance was worst generally when using `T` rather than just having `::Function` fields for example.

Fine to leave this note in place though.

Reply:
Hmm, interesting. I wonder if we're talking past each other here a little bit. What I mean in that note is something like `LinkRule{T}`, and then you'd `enable!` a fully qualified instance of it (i.e. `enable!(p, LinkRule{Foo}()`). I don't see how that would really affect performance, since I don't think `p.rules` is even really accessed during parsing?

I presume you might mean having the `Rule` type take type parameters?

Reply:
> I presume you might mean having the Rule type take type parameters?

Either really, I'm not seeing a use case for where type parameters would really be needed for how the rules are meant to be defined/used. There may very well be one that I'm not seeing, but having the note there is fine by me. We can revisit should be come to a situation where type parameters are actually required somewhere to make things work nicely.